impeller/entity/contents/linear_gradient_contents.cc

```diff
@@ -53,9 +54,91 @@ bool LinearGradientContents::IsOpaque() const {
   return true;
 }
 
+// A much faster (in terms of ALU) linear gradient that uses vertex
+// interpolation to perform all color computation. Requires that the geometry of
+// the gradient is divided into regions based on the stop values.
+// Currently restricted to rect geometry where the start and end points are
+// perfectly horizontal/vertical, but could easily be expanded to StC cases
+// provided that the start/end are on our outside the coverage rect.
+bool LinearGradientContents::FastLinearGradient(const ContentContext& renderer,
+                                                const Entity& entity,
+                                                RenderPass& pass) const {
+  using VS = GradientPipeline::VertexShader;
+  using FS = GradientPipeline::FragmentShader;
+
+  auto options = OptionsFromPassAndEntity(pass, entity);
+  options.primitive_type = PrimitiveType::kTriangle;
+  Geometry& geometry = *GetGeometry();
+
+  // We already know this is an axis aligned rectangle, so the coverage will
+  // be approximately the same as the geometry. For non AARs, we can force
```

Comment:
Just say "axis aligned rectangles" instead of AARs?

Reply:
Done